File: 2021/Day 1/C++/part1.cpp
#include <fstream>
#include <iostream>
#include <string>
#include <vector>

using namespace std;

int main() {
    // Read data into a vector
    vector<int> numbers;
    ifstream datafile;
    datafile.open("../data.txt");
    if (datafile.is_open()) {
        string line;
        while (datafile.good()) {
            getline(datafile, line);
            numbers.push_back(stoi(line));
        }
    } else {
        cout << "Could not open file." << endl;
    }
    // Iterate over data to find count
    int prev = numbers[0];
    int largerCount = 0;
    for (auto number : numbers) {
        if (number > prev) largerCount++;
        prev = number;
    }
    cout << "Larger count: " << largerCount << endl;
}

File: 2021/Day 1/JS/part1.ts
import * as fs from 'fs'
import path from 'path'

// Read data
const file = fs.readFileSync(path.resolve(__dirname, '../data.txt'), 'utf8')
const numbers = file.split('\n')

// Count times the number increased
let countLarger = 1
for (let i = 1; i < numbers.length; i++) {
    if (numbers[i] > numbers[i - 1]) countLarger++
}

// Output answer
console.log('Count larger:', countLarger)
